geoips/commandline/geoips_command.py

```diff
@@ -0,0 +1,357 @@
+
+        Initialize the plugin packages and their paths. This is done by using the
+        get_plugin_packages() and get_plugin_package_paths() functions.
+        """
+        self.entrypoints = [
+            ep.value for ep in get_entry_point_group("geoips.plugin_packages")
+        ]
+        self.paths = [
+            dirname(resources.files(ep.value))
+            for ep in get_entry_point_group("geoips.plugin_packages")
+        ]
+
+
+plugin_packages = PluginPackages()
+
+
+class GeoipsCommand(abc.ABC):
+    """Abstract Base Class for top-level GeoIPS Command Classes, such as get or list.
+
+    This class is a blueprint of what each top-level GeoIPS Command Classes should
+    implement. Includes shared attributes and an ``add_suparsers`` function which is
+    used for initializing sub-command classes of a certain GeoIPS Command.
+    """
+
+    def __init__(self, parent=None):
+        """Initialize GeoipsCommand with a subparser and default to the command func.
+
+        Do this for each GeoipsCLI.geoips_subcommand_classes. This will instantiate
+        each subcommand class with a parser and point towards the correct default
+        function to call if that subcommand has been called.
+
+        Parameters
+        ----------
+        parent: optional - GeoipsCommand Class
+            - The parent command class that possibly is initializing it's child.
+              Ex. GeoipsList would invoke this init function for each of its subcommand
+              classes (GeoipsListPackages, GeoipsListScripts, ...). When it invokes this
+              init, it supplies 'self' as an argument to follow the correct logic below.
+        """
+        self.github_org_url = "https://github.com/NRLMMD-GEOIPS/"
+        if parent:
+            # Set the combined name of the provided object. For example, if this was the
+            # parent 'list' command, it would be 'geoips_list'. If it was a child of
+            # list, for example 'scripts', combined name would be 'geoips_list_scripts'
+            self.combined_name = f"{parent.combined_name}_{self.command_name}"
+            if parent.cmd_instructions:
+                # this is used for testing purposes to ensure failure for invalid
+                # help information. If the parent already has cmd_instructions set,
+                # use these instructions so we can test proper functionality of the CLI.
+                self.cmd_instructions = parent.cmd_instructions
+            else:
+                # Otherwise use the default cmd_instructions which are used for normal
+                # invocation of the CLI.
+                self.cmd_instructions = cmd_instructions
+            try:
+
+                # attempt to create a sepate sub-parser for the specific sub-command
+                # class being initialized
+                # So we can separate the commands arguments in a tree-like structure
+                self.subcommand_parser = parent.subparsers.add_parser(
```

Comment:
Given the change that we made to go from `subcommand_name` to `command_name`, this should probably be `command_parser` rather than `subcommand_parser`, right?

Reply:
Agreed. I can do that now or create an issue for it. Which do you prefer @jsolbrig?

Reply:
Let's create an issue for this so that we can get this merged. This is a minor, non-functional change that could impact down-stream PRs.